### PR TITLE
Fixed CollectionView SelectedItem binding issue on initial loading

### LINKED ISSUE #23299: [MAUI] I4_ Selection - The default items not highlight in Single Pre-selection and Multiple Pre-selection Page
### Description

Affected case:
Single Pre-selection (Vertical list)
Multiple Pre-selection (Vertical list)

### Steps to Reproduce

**Steps to Reproduce**

1. Deploy https://devdiv.visualstudio.com/Pkgs/_git/VS.TestAssets.Xaml?path=/Assets/Projects/ManualMauiTests to Windows.
2. Navigate to 'I CollectionView' ->'I4 Selection '
3. Open Single Pre-selection or Multiple Pre-selection Page
4. Check the default items should be highlighted

**Actual Behavior:**
image

**Expected Behavior:**
The fourth item should be highlighted by default. 

### Link to public reproduction project repository

https://devdiv.visualstudio.com/Pkgs/_git/VS.TestAssets.Xaml?path=/Assets/Projects/ManualMauiTests

### Version with bug
Microsoft.Maui.Controls 8.0.70-ci.net8.24325.2

### Is this a regression from previous behavior?
Not a regression, The issue also repro on 8.0.61& 8.0.3

### Affected platforms

iOS, Windows, macOS

### LINKED ISSUE #25551: CollectionView Programmatic-Selection is not updated when binding the selectedItem
### Description
In CollectionView, when selecting through Programmatic selection is not applied. 

### Steps to Reproduce

Deploy the attached sample
By default, SelectedItem is bound to the string on code behind
Check the default items should be highlighted

### Actual Behavior:
No item is selected on initial loading 
Image

### Expected Behavior:
Item should be selected and Highlighted
Image

### Link to public reproduction project repository
[CollectionView_selection.zip](https://github.com/user-attachments/files/17543427/CollectionView_selection.zip)

### Version with bug

9.0.0-rc.2.24503.2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Windows, macOS

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by SuthiYuvaraj**
### Issue Details
1. **Selected Item Binding Issue**: The selected item binding does not work as expected across Windows, iOS, and Mac platforms.
2. **Visual State Issue**: If a selection visual state is present, it is not applied to the view.
 
### Root Cause
- **Windows**: The selected item is reset due to the `ListViewBase.IsSynchronizedWithCurrentItem` property setting. Additionally, the visual state is applied before the item view control is created.
- **iOS & Mac**: The selected index is not applied during initial loading, specifically after updating the `ItemsSource`.
 
### Fix Details
- **Windows**: The existing local variable is used to prevent resetting `SelectedItem` while updating the native view property. Additionally, the `ItemContentControl`'s visual state is now applied after it is created.
- **iOS & Mac**: The selected index is applied within a batch update to ensure that items are generated before applying selection.

Reference:
iOS - [ios - How to select some items in the UICollectionView after it first load? - Stack Overflow](https://stackoverflow.com/questions/15228472/how-to-select-some-items-in-the-uicollectionview-after-it-first-load)

### Issues Fixed
Fixes #25551 
Fixes #23299

### Output Screenshot
Before Issue Fix | After Issue Fix |
|----------|----------|
|<image width="600" height="300" alt="Before Fix video" src="https://github.com/user-attachments/assets/76ad40bf-d654-4684-a57b-085254017507">|<image width="600" height="300" alt="After Fix video"  src="https://github.com/user-attachments/assets/590f8ee9-0f56-464f-84cf-467aa157efe6" >|
|<image width="600" height="300" alt="Before Fix video" src="https://github.com/user-attachments/assets/5337b0a8-9895-4183-9d25-ec8484cb23f9">|<image width="600" height="300" alt="After Fix video"  src="https://github.com/user-attachments/assets/493e5b26-d1e5-4bc4-8646-3ce191772654" >|

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25551.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		{
+		}
+
+		[Test]
+		[Category(UITestCategories.CollectionView)]
+		
+		public void CollectionViewPreSelectionShouldUpdate()
+		{
+			App.WaitForElement("ListCollection");
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25551.cs:27]: There are some pending screenshots (Android etc). You can pick them from the latest build if are correct.

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25551.cs (hunk 1 above)
Comment (SuthiYuvaraj) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25551.cs:27]: @jsuarezruiz , Thank you for the feedback, I have retrieved the images from the recent build and updated the PR accordingly. Please review and let me know if there are any additional concerns.

**Review by PureWeen — CHANGES_REQUESTED** [review]
CollectionViewPreSelectionShouldUpdate

Failing

**Comment by SuthiYuvaraj** [comment]
Hi @PureWeen  , The **CollectionViewPreSelectionShouldUpdate** failed due to a missing snapshot on the Windows platform. We have now included the snapshot. Please review and let us know if there are any concerns.

**Review by rmarinho — APPROVED** [review]

**Pull request merged**